Modules/System/AL Rest Client/src/ALHttpContent.Codeunit.al

```diff
@@ -0,0 +1,261 @@
+/// <summary>Holder object for the Http Content data.</summary>
+codeunit 2354 "AL Http Content"
+{
+    var
+        HttpContent: HttpContent;
+        ContentTypeEmptyErr: Label 'The value of the Content-Type header must be specified.';
+        ContentNotSupportedErr: Label 'The content is not supported.';
+        MimeTypeTextPlainTxt: Label 'text/plain', Locked = true;
+        MimeTypeTextXmlTxt: Label 'text/xml', Locked = true;
+        MimeTypeApplicationOctetStreamTxt: Label 'application/octet-stream', Locked = true;
+        MimeTypeApplicationJsonTxt: Label 'application/json', Locked = true;
+
+    #region Constructors
+    /// <summary>Initializes a new instance of the ALHttpContent class with the specified content.</summary>
+    /// <param name="Content">The content to send to the server. Valid types are Codeunit (Temp Blob), InStream, JsonObject, JsonArray, JsonToken, XmlDocument, Text</param>
+    /// <returns>The AL Http Content object</returns>
+    /// <remarks>
+    /// Text types will automatically get 'text/plain'.
+    /// Json types will automatically get 'application/json'. 
+    /// Xml types will automatically get 'text/xml'. 
+    /// Codeunit "Temp Blob" and InStream types will automatically get 'application/octet-stream'.
+    /// </remarks>
+
+    procedure Create(Content: Variant) ALHttpContent: Codeunit "AL Http Content";
+    begin
+        ALHttpContent := Create(Content, '');
+    end;
+
+    /// <summary>Initializes a new instance of the ALHttpContent class with the specified content and content type.</summary>
+    /// <param name="Content">The content to send to the server. Valid types are Codeunit (Temp Blob), InStream, JsonObject, JsonArray, JsonToken, XmlDocument, Text</param>
+    /// <param name="ContentType">The content type of the content to send to the server.</param>
+    /// <returns>The AL Http Content object.</returns>
+    /// <remarks>
+    /// Text types will automatically get 'text/plain' if not specified in the ContentType parameter.
+    /// Json types will automatically get 'application/json'. 
+    /// Xml types will automatically get 'text/xml'. 
+    /// Codeunit "Temp Blob" and InStream types will automatically get 'application/octet-stream' if not specified in the ContentType parameter.
+    /// </remarks>
+    procedure Create(Content: Variant; ContentType: Text) ALHttpContent: Codeunit "AL Http Content"
```

Comment:
I know it's more tedious, but I think it would be better to have an overload for every supported type instead of accepting a `Variant`. This way developers would see an error at compile time instead of run time if they attempt to pass something unsupported.

Reply:
I agree, the Create method has now been replaced by overloads for every supported type.

Reply:
But it also means that the Send method on codeunit "Rest Client" needs to have the same overloads.